src/auth.js

```diff
@@ -58,6 +41,21 @@ const IncludeGrantedScopes = ['none', 'user', 'team'];
 */
 export default class DropboxAuth {
   constructor(options) {
+    // Setup Globals
+    if (isBrowserEnv()) {
+      fetch = window.fetch.bind(window);
+      crypto = window.crypto || window.msCrypto; // for IE11
+    } else {
+      fetch = require('node-fetch'); // eslint-disable-line global-require
+      crypto = require('crypto'); // eslint-disable-line global-require
+    }
+
+    if (typeof TextEncoder === 'undefined') {
+      Encoder = require('util').TextEncoder; // eslint-disable-line global-require
+    } else {
+      Encoder = TextEncoder;
+    }
+
```

Comment:
How does this ensure `window` is defined?

Nit: this runs each time the constructor runs. I don't think it should matter in terms of behavior, but it would be cleaner to just wrap these around "getter" methods. Or have them set up in a separate function which is only allowed to be called once. Or have the constructor set local properties for each, not just `fetch`.

Reply:
So this covers a specific case where window is not defined when the class is imported but is defined when code is actually run. You can look at the linked issue to see the context from it.